resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreaker.java

```diff
@@ -556,6 +555,16 @@ static CircuitBreaker of(String name,
      */
     void onSuccess(long duration, TimeUnit durationUnit);
```

Comment:
Doesn't need to deprecate `onSuccess`? I think `onResult` can cover `onSuccess`.

Reply:
I wasn't sure. I don't really like methods where you have to provide a `null` parameter, because it's optional. I prefer to use Method Overloading.
But I thought giving the overloaded method a name like `onResult` explains the intention better than `onSuccess`.

Reply:
Understood 👍

Reply:
but maybe we still need to add @deprecated so we can move it from version 2 if it is not still needed ?

Reply:
It's still needed. There are some functions which can be decorated and don't return a value. Like Runnable. Or types like Completable, Maybe.